### PR TITLE
server/sandbox: fix add/release pod/ctr ids and names

### LINKED ISSUE #127: Container ID does not exist error...
Repro:

Terminal 1:
`sudo ocid --debug --runtime $(which runc)`

Terminal 2:
```
sudo ocic pod create --config test/testdata/sandbox_config.json
sudo ocic ctr create --pod ca28917626e92b123e6ca09180b5f13feea012da925841584b0b255b55b4056b --config test/testdata/container_redis.json
```

Terminal 1 output:
```
INFO[0256] pod container state &{{ default-podsandbox1-0-infra running 8605 /var/lib/ocid/sandboxes/ca28917626e92b123e6ca09180b5f13feea012da925841584b0b255b55b4056b map[ocid/labels:{"group":"test"} ocid/log_path:. ocid/name:default-podsandbox1-0 owner:hmeng ocid/annotations:{"owner":"hmeng"} ocid/container_id:8d33f6c911e54f6e8091b67b8d62577c725b0b7b467f421a5b648c796da0717d ocid/container_name:default-podsandbox1-0-infra]} 2016-10-12 18:57:42.943754326 +0000 UTC 2016-10-12 13:57:43.013389411 -0500 CDT 0001-01-01 00:00:00 +0000 UTC 0} 
...
           _.-``__ ''-._                                             
      _.-``    `.  `_.  ''-._           Redis 3.2.4 (00000000/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._                                   
 (    '      ,       .-`  | `,    )     Running in standalone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 1
  `-._    `-._  `-./  _.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |           http://redis.io 
```

Container list shows two containers after creating one:
```
sudo ocic ctr list
ID: 629ef4181d4ceb0dcb54198fad1dba2f4dbbc5827ab0f91c332a5be1512d3861
Pod: ca28917626e92b123e6ca09180b5f13feea012da925841584b0b255b55b4056b
Status: CREATED
Created: 2016-10-12 14:01:45 -0500 CDT
ID: 8d33f6c911e54f6e8091b67b8d62577c725b0b7b467f421a5b648c796da0717d
Pod: ca28917626e92b123e6ca09180b5f13feea012da925841584b0b255b55b4056b
Status: RUNNING
Created: 2016-10-12 13:57:42 -0500 CDT
```
Can get status for the first container...:
```
sudo ocic ctr status --id 629ef4181d4ceb0dcb54198fad1dba2f4dbbc5827ab0f91c332a5be1512d3861
ID: 629ef4181d4ceb0dcb54198fad1dba2f4dbbc5827ab0f91c332a5be1512d3861
Status: RUNNING
Created: 2016-10-12 14:01:45 -0500 CDT
Started: 2016-10-12 14:03:51 -0500 CDT
```

**FATA(L) Error** - Can't get status for the second container:
```
sudo ocic ctr status --id 8d33f6c911e54f6e8091b67b8d62577c725b0b7b467f421a5b648c796da0717d
FATA[0000] Getting the status of the container failed: rpc error: code = 2 desc = container with ID starting with 8d33f6c911e54f6e8091b67b8d62577c725b0b7b467f421a5b648c796da0717d not found: ID does not exist 
```

runc list
```
sudo runc list
ID                                          PID         STATUS      BUNDLE                                                                                      CREATED
default-podsandbox1-0-infra                 8605        running     /var/lib/ocid/sandboxes/ca28917626e92b123e6ca09180b5f13feea012da925841584b0b255b55b4056b    2016-10-12T18:57:42.943754326Z
default-podsandbox1-0-podsandbox1-redis-0   8724        running     /var/lib/ocid/containers/629ef4181d4ceb0dcb54198fad1dba2f4dbbc5827ab0f91c332a5be1512d3861   2016-10-12T19:01:45.51429181Z
```
runc state:
```
sudo runc state default-podsandbox1-0-infra
{
  "ociVersion": "1.0.0-rc1-dev",
  "id": "default-podsandbox1-0-infra",
  "pid": 8605,
  "bundlePath": "/var/lib/ocid/sandboxes/ca28917626e92b123e6ca09180b5f13feea012da925841584b0b255b55b4056b",
  "rootfsPath": "/var/lib/ocid/graph/vfs/pause/rootfs",
  "status": "running",
  "created": "2016-10-12T18:57:42.943754326Z",
  "annotations": {
    "ocid/annotations": "{\"owner\":\"hmeng\"}",
    "ocid/container_id": "8d33f6c911e54f6e8091b67b8d62577c725b0b7b467f421a5b648c796da0717d",
    "ocid/container_name": "default-podsandbox1-0-infra",
    "ocid/labels": "{\"group\":\"test\"}",
    "ocid/log_path": ".",
    "ocid/name": "default-podsandbox1-0",
    "owner": "hmeng"
  }
}

sudo runc state default-podsandbox1-0-podsandbox1-redis-0
{
  "ociVersion": "1.0.0-rc1-dev",
  "id": "default-podsandbox1-0-podsandbox1-redis-0",
  "pid": 8724,
  "bundlePath": "/var/lib/ocid/containers/629ef4181d4ceb0dcb54198fad1dba2f4dbbc5827ab0f91c332a5be1512d3861",
  "rootfsPath": "/var/lib/ocid/containers/629ef4181d4ceb0dcb54198fad1dba2f4dbbc5827ab0f91c332a5be1512d3861/rootfs",
  "status": "running",
  "created": "2016-10-12T19:01:45.51429181Z",
  "annotations": {
    "ocid/labels": "{\"tier\":\"backend\"}",
    "ocid/log_path": "container.log",
    "ocid/name": "default-podsandbox1-0-podsandbox1-redis-0",
    "ocid/sandbox_id": "ca28917626e92b123e6ca09180b5f13feea012da925841584b0b255b55b4056b",
    "ocid/tty": "false",
    "pod": "podsandbox1"
  }
}
```

### PR DESCRIPTION
**Pull request by runcom**
Fix #127

@mrunalp @mikebrow PTAL

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
can't sign the CLA right now since I made the request with an old, wrong email address, I contacted the linux foundation, still waiting..

**Comment by runcom** [comment]
alright, fixed

**Comment by runcom** [comment]
alright, fixed

File: test/ctr.bats [hunk 1]
```diff
@@ -108,3 +108,36 @@ function teardown() {
 	cleanup_pods
 	stop_ocid
 }
+
+@test "ctr status #127" {
```
Comment (runcom) [inline test/ctr.bats]: @mrunalp thought it would be nice to have references to issue fixed by a particular regression test

Comment (mrunalp) [inline test/ctr.bats]: I think that adding that info in commit message should be discoverable. Also, we could add comments in the code referring to an issue if it is handling some particularly hairy corner case. A description like "all ctrs in pod status" should work, IMO.

Comment (runcom) [inline test/ctr.bats]: updated

**Comment by mrunalp** [comment]
Looks good. I will test out locally and merge.

**Comment by mikebrow** [comment]
Looks better but after `sudo ocic pod remove` does not delete both containers from the /var/lib/ocid/containers directory. Thus trying to run my manual test twice results in a failure the second time around.

**Comment by mikebrow** [comment]
```
sudo ocic ctr list
ID: 6a487097ff18d03ac02e55d6ee73363288e1c92011ed02668fd93426dcf44165
Pod: 1035d00d5a2e8f2c37307eed1c493773f3704c1e40a3857d7e90d6a5c73edf67
Status: RUNNING
Created: 2016-10-13 11:28:36 -0500 CDT
ID: 617f2c3dd898356db711b980bf85567a9b718cb3378ab72eb64c8352c3dd5647
Pod: 1035d00d5a2e8f2c37307eed1c493773f3704c1e40a3857d7e90d6a5c73edf67
Status: RUNNING
Created: 2016-10-13 11:28:00 -0500 CDT
mike@mike-VirtualBox:~/go/src/github.com/kubernetes-incubator/cri-o$ sudo ocic ctr status --id 6a487097ff18d03ac02e55d6ee73363288e1c92011ed02668fd93426dcf44165
ID: 6a487097ff18d03ac02e55d6ee73363288e1c92011ed02668fd93426dcf44165
Status: RUNNING
Created: 2016-10-13 11:28:36 -0500 CDT
Started: 2016-10-13 11:29:19 -0500 CDT
mike@mike-VirtualBox:~/go/src/github.com/kubernetes-incubator/cri-o$ sudo ocic ctr status --id 617f2c3dd898356db711b980bf85567a9b718cb3378ab72eb64c8352c3dd5647
ID: 617f2c3dd898356db711b980bf85567a9b718cb3378ab72eb64c8352c3dd5647
Status: RUNNING
Created: 2016-10-13 11:28:00 -0500 CDT
Started: 2016-10-13 11:28:00 -0500 CDT
mike@mike-VirtualBox:~/go/src/github.com/kubernetes-incubator/cri-o$ sudo ocic pod list
ID: 1035d00d5a2e8f2c37307eed1c493773f3704c1e40a3857d7e90d6a5c73edf67
Name: podsandbox1
Status: READY
Created: 2016-10-13 11:28:00 -0500 CDT
Labels:
	group -> test
Annotations:
	owner -> hmeng
mike@mike-VirtualBox:~/go/src/github.com/kubernetes-incubator/cri-o$ sudo ocic pod remove --id 1035d00d5a2e8f2c37307eed1c493773f3704c1e40a3857d7e90d6a5c73edf67
1035d00d5a2e8f2c37307eed1c493773f3704c1e40a3857d7e90d6a5c73edf67
mike@mike-VirtualBox:~/go/src/github.com/kubernetes-incubator/cri-o$ sudo runc list
ID          PID         STATUS      BUNDLE      CREATED

mike@mike-VirtualBox:/var/lib/ocid$ ls containers
6a487097ff18d03ac02e55d6ee73363288e1c92011ed02668fd93426dcf44165
```

**Comment by runcom** [comment]
> Looks better but after sudo ocic pod remove does not delete both containers from the /var/lib/ocid/containers directory. Thus trying to run my manual test twice results in a failure the second time around.

I'll look into this as well...

**Comment by runcom** [comment]
seems that issue is wanted https://github.com/kubernetes-incubator/cri-o/blob/master/server/sandbox.go#L348 @mrunalp?

**Comment by mrunalp** [comment]
@runcom The pod infra container directory is removed after all the other containers in the pod have been removed. https://github.com/kubernetes-incubator/cri-o/blob/master/server/sandbox.go#L365. We can open and fix the issue @mikebrow reported separately.

**Comment by mikebrow** [comment]
Trying to keep the pause container alive?

**Comment by runcom** [comment]
> @runcom The pod infra container directory is removed after all the other containers in the pod have been removed. https://github.com/kubernetes-incubator/cri-o/blob/master/server/sandbox.go#L365. We can open and fix the issue @mikebrow reported separately.

I have fixed it here, it was just Name vs ID, PTAL

**Comment by runcom** [comment]
@mikebrow please re-test

**Comment by mrunalp** [comment]
@runcom Nice! :)

**Comment by mikebrow** [comment]
mike@mike-VirtualBox:~/go/src/github.com/kubernetes-incubator/cri-o$ sudo ocic pod create --config test/testdata/sandbox_config.jsonabadb43d94f84e389406ab50310b6053062976e61980566bc75301d2857da71e
mike@mike-VirtualBox:~/go/src/github.com/kubernetes-incubator/cri-o$ sudo ocic ctr create --pod abadb43d94f84e389406ab50310b6053062976e61980566bc75301d2857da71e --config test/testdata/container_redis.json
FATA[0001] Creating container failed: rpc error: code = 2 desc = error getting container state for default-podsandbox1-0-podsandbox1-redis-0: wait: no child processes: [123 10 32 32 34 111 99 105 86 101 114 115 105 111 110 34 58 32 34 49 46 48 46 48 45 114 99 49 45 100 101 118 34 44 10 32 32 34 105 100 34 58 32 34 100 101 102 97 117 108 116 45 112 111 100 115 97 110 100 98 111 120 49 45 48 45 112 111 100 115 97 110 100 98 111 120 49 45 114 101 100 105 115 45 48 34 44 10 32 32 34 112 105 100 34 58 32 49 54 49 52 53 44 10 32 32 34 98 117 110 100 108 101 80 97 116 104 34 58 32 34 47 118 97 114 47 108 105 98 47 111 99 105 100 47 99 111 110 116 97 105 110 101 114 115 47 49 100 54 101 49 53 52 50 54 101 51 54 102 99 55 57 55 102 99 50 100 50 52 52 57 55 55 57 56 53 50 98 52 57 99 50 57 55 49 100 101 51 99 53 101 50 49 54 56 53 48 52 99 52 97 57 98 57 97 51 56 57 55 49 34 44 10 32 32 34 114 111 111 116 102 115 80 97 116 104 34 58 32 34 47 118 97 114 47 108 105 98 47 111 99 105 100 47 99 111 110 116 97 105 110 101 114 115 47 49 100 54 101 49 53 52 50 54 101 51 54 102 99 55 57 55 102 99 50 100 50 52 52 57 55 55 57 56 53 50 98 52 57 99 50 57 55 49 100 101 51 99 53 101 50 49 54 56 53 48 52 99 52 97 57 98 57 97 51 56 57 55 49 47 114 111 111 116 102 115 34 44 10 32 32 34 115 116 97 116 117 115 34 58 32 34 99 114 101 97 116 101 100 34 44 10 32 32 34 99 114 101 97 116 101 100 34 58 32 34 50 48 49 54 45 49 48 45 49 51 84 49 54 58 53 50 58 49 57 46 56 57 48 49 54 54 51 54 55 90 34 44 10 32 32 34 97 110 110 111 116 97 116 105 111 110 115 34 58 32 123 10 32 32 32 32 34 111 99 105 100 47 108 97 98 101 108 115 34 58 32 34 123 92 34 116 105 101 114 92 34 58 92 34 98 97 99 107 101 110 100 92 34 125 34 44 10 32 32 32 32 34 111 99 105 100 47 108 111 103 95 112 97 116 104 34 58 32 34 99 111 110 116 97 105 110 101 114 46 108 111 103 34 44 10 32 32 32 32 34 111 99 105 100 47 110 97 109 101 34 58 32 34 100 101 102 97 117 108 116 45 112 111 100 115 97 110 100 98 111 120 49 45 48 45 112 111 100 115 97 110 100 98 111 120 49 45 114 101 100 105 115 45 48 34 44 10 32 32 32 32 34 111 99 105 100 47 115 97 110 100 98 111 120 95 105 100 34 58 32 34 97 98 97 100 98 52 51 100 57 52 102 56 52 101 51 56 57 52 48 54 97 98 53 48 51 49 48 98 54 48 53 51 48 54 50 57 55 54 101 54 49 57 56 48 53 54 54 98 99 55 53 51 48 49 100 50 56 53 55 100 97 55 49 101 34 44 10 32 32 32 32 34 111 99 105 100 47 116 116 121 34 58 32 34 102 97 108 115 101 34 44 10 32 32 32 32 34 112 111 100 34 58 32 34 112 111 100 115 97 110 100 98 111 120 49 34 10 32 32 125 10 125]

**Comment by mikebrow** [comment]
not sure what's up rebooting..

**Comment by mrunalp** [comment]
@mikebrow Looks like another issue. Could you open one for it? (Thanks for testing and finding issues btw :) )

**Comment by runcom** [comment]
definitely not related? bad state of /var/lib/ocid/* or runc

**Comment by mrunalp** [comment]
LGTM

**Comment by mikebrow** [comment]
Not sure what's up but now I can't create a container.
```
mike@mike-VirtualBox:~/go/src/github.com/kubernetes-incubator/cri-o$ sudo ocic pod create --config test/testdata/sandbox_config.json
[sudo] password for mike: 
99fec541a30f381f55c51c8cf4a384294153321f808b8e1c61f1b4efe93f1916
mike@mike-VirtualBox:~/go/src/github.com/kubernetes-incubator/cri-o$ sudo ocic ctr create --pod 99fec541a30f381f55c51c8cf4a384294153321f808b8e1c61f1b4efe93f1916 --config test/testdata/container_redis.json
FATA[0008] Creating container failed: rpc error: code = 2 desc = `docker rm 080880564dae577b4518de756aa82742c59f7705032c186428b7fa692412180e` failed:  (wait: no child processes) 
```

**Comment by runcom** [comment]
@mikebrow seems like you have a bad state between docker/runc/ocid. Can you purge everything and retry from scratch? (open an issue also)

**Comment by runcom** [comment]
@mrunalp would you merge this? :angel:

**Comment by mikebrow** [comment]
Sure what do I need to completely purge.. i deleted the stuff under /var/lib/ocid/containers/* and sandboxes/* and checked to make sure nothing was running via docker ps and runc...

**Comment by runcom** [comment]
> Sure what do I need to completely purge.. i deleted the stuff under /var/lib/ocid/containers/* and sandboxes/* and checked to make sure nothing was running via docker ps and runc...

the failure you had seems related to docker itself, docker rm is failing I don't know why.. we hopefully get rid of docker soon also

**Comment by mikebrow** [comment]
Ok .. now it's working. So some sort of intermittent problems. The first one looked like processes were leaking so I ran out of pids.. The second one looks like some odd timing bug.

So for this one LGTM :-)
